### DISCUSSION TITLE
Docker for Arm (Raspberry Pi)

### DISCUSSION THREAD

**Discussion by darrepac** [abraunegg/onedrive#1409] [Q&A] [answered]
Hi

I probably missed something but I do not find the Arm version of docker.
Something similar than `docker pull driveone/onedrive:stretch` but for Arm

thx

**Comment by abraunegg** [comment, marked as answer]
@darrepac 
You need to build your ARM Docker image yourself. Refer to: https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#building-a-custom-docker-image

**Reply by darrepac** [reply]
Oh

**Reply by abraunegg** [reply]
Additionally pay extreme attention to the build requirements: https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#build-environment-requirements

**Reply by darrepac** [reply]
Well, will try the package, seems there is one

**Reply by abraunegg** [reply]
Do not use **any** package .. It will be the incorrect package - refer to: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#installing-from-distribution-packages-or-building-the-onedrive-client-for-linux-from-source

**Reply by darrepac** [reply]
yes I read it and it doesn't say to not use debian (Raspbian is based on it) package
When we click on the debian link in your table, the package refer to 2.4.10-1 version which seems not so old...

All in all, if it is not safe and if there is no package neither docker version, should I consider that raspberry is not a good platform to run Onedrive client (I thought it would need minimal ressource but I may be wrong)

**Reply by abraunegg** [reply]
Rasbian packages are actually very old last time I checked.

Raspberry Pi is a good platform for the client - that you cant find a Docker / package to install directly, this can be improved .. perhaps consider contributing back to the open source community.

Building the client or Docker image for ARM is also not hard or time intensive ...

**Reply by darrepac** [reply]
ok if you confirm that the platform is ok, I may give a try

**Reply by darrepac** [reply]
I tried 2 times and it crashes my Pi two times... so I will find an other way. Thanks for your support

**Reply by abraunegg** [reply]
@darrepac 
> I tried 2 times and it crashes my Pi two times...

What crashed? the compile or the Pi ?

As I advised - you need to pay attention to the build requirements. This is the *main* cause why it crashes / fails to build. You need:
* 1 GB Memory
* 1 GB or more of Swapspace

On Rasbian, Swapspace defaults to 256 MB. Unless you fix that, reboot, double check you now have >1024 MB swap space available - any build of the client or Docker build will fail.

99.99% of the time when people have run into issues on a Pi it is because of this reason.
